cpp/fsm.cc

```diff
@@ -446,7 +446,7 @@ class CompactFSM::Impl : public FSMImplBase<Compact2DArray<FSMEdge>> {
  public:
   using FSMImplBase<Compact2DArray<FSMEdge>>::FSMImplBase;
 
-  int GetNextState(int from, int value, EdgeType edge_type) const;
+  void GetNextStates(int from, int value, std::vector<int>& targets, EdgeType edge_type) const;
```

Comment:
```suggestion
  void GetNextStates(int from, int value, EdgeType edge_type, std::vector<int>* target) const;
```
Pass editable params by pointer; put at last